libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltDatasetCohort.ts

```diff
@@ -0,0 +1,199 @@
+          }
+          case CohortColumnNames.TrueY: {
+            const filter = translateCohortFilterForTarget(
+              preBuiltCohortFilter,
+              props.dataset,
+              modelType,
+              CohortColumnNames.TrueY
+            );
+            filterList.push(filter);
+            break;
+          }
+          case CohortColumnNames.ClassificationOutcome: {
+            const filter = translateCohortFilterForClassificationOutcome(
+              preBuiltCohortFilter,
+              featureRanges
+            );
+            filterList.push(filter);
+            break;
+          }
+          default: {
+            const filter = translateCohortFilterForDataset(
+              preBuiltCohortFilter,
+              featureRanges
+            );
+            if (filter) {
+              filterList.push(filter);
+            }
+            break;
+          }
+        }
+      });
+      const datasetCohort = new DatasetCohort(
+        preBuiltCohort.name,
+        props.dataset,
+        filterList,
+        modelType,
+        featureRanges,
+        CohortSource.Prebuilt
+      );
+      datasetCohorts.push(datasetCohort);
+    }
+  }
+  return datasetCohorts;
+}
+
+function translateCohortFilterForTarget(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  dataset: IDataset,
+  modelType: ModelTypes,
+  cohortColumn: CohortColumnNames
+): IFilter {
+  const filterColumn =
+    cohortColumn === CohortColumnNames.TrueY
+      ? DatasetCohortColumns.TrueY
+      : DatasetCohortColumns.PredictedY;
+  if (IsClassifier(modelType)) {
+    const modelClasses = dataset.class_names;
+    if (modelClasses) {
+      const index = preBuiltCohortFilter.arg
+        .map((modelClass) => modelClasses?.indexOf(modelClass))
+        .filter((index) => index !== -1);
+      index.sort((a, b) => a - b);
+      const filter: IFilter = {
+        arg: index,
+        column: cohortColumn,
+        method: preBuiltCohortFilter.method
+      };
+      return filter;
+    }
+  }
+  const filter: IFilter = {
+    arg: preBuiltCohortFilter.arg,
+    column: filterColumn,
+    method: preBuiltCohortFilter.method
+  };
+  return filter;
+}
+
+function translateCohortFilterForClassificationOutcome(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  featureRanges: {
+    [key: string]: INumericRange | ICategoricalRange;
+  }
+): IFilter {
+  let index: number[] = [];
+  const featureRange = featureRanges[DatasetCohortColumns.ClassificationError];
+  if (featureRange) {
+    const allowedClassificationErrorValues = (featureRange as ICategoricalRange)
+      .uniqueValues;
+    index = preBuiltCohortFilter.arg
+      .map((classificationError) =>
+        allowedClassificationErrorValues.indexOf(classificationError)
+      )
+      .filter((index) => index !== -1);
+  }
+  index.sort((a, b) => a - b);
+  const filter: IFilter = {
+    arg: index,
+    column: DatasetCohortColumns.ClassificationError,
+    method: preBuiltCohortFilter.method
+  };
```

Comment:
what do you think about extracting this out to a small method that takes in arg, column as parameters and returns a method filter?  Like "createMethodFilter"?
It could be re-used in several places in this file - lines 161, 135, 127, 39, 48, 190
So it could reduce a lot of lines in this file?

Reply:
thanks Ilya,  I think we also need to pass in preBuiltCohortFilter.method, as the third parameters? or are you suggesting to create a new filter type? please let me know if you have other thoughts
and by extracting this out to a small method by taking those three params and return a IFilter. it  is actually taking more lines, as there will be one more function and the value for column and method are too long, when leveraging the function, it still takes 5 lines everytime. Overall the file increases 8 lines.

Reply:
merged the change, plz let me know if any thoughts, I can always update thanks!